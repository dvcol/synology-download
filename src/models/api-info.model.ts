export interface ApiInfo {
  path: string;
  minVersion: number;
  maxVersion: number;
}
