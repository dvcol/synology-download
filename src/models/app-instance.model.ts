export enum AppInstance {
  option = 'synology-download-option',
  popup = 'synology-download-popup',
  content = 'synology-download-content',
  standalone = 'synology-download-standalone',
}
