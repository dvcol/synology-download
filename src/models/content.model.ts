import type { TabCount, TabStatus } from './tab.model';

export enum ContentSource {
  Task = 'task',
  Download = 'download',
}

export interface Content {
  source: ContentSource;
  id: string | number;
  key: string;
  status: TabStatus;
  title?: string;
  folder?: string;
  progress?: number;
  speed?: number;
  size?: number;
  received?: number;
  eta?: string;
  createdAt?: number;
  finishedAt?: number;
}

/**
 * Enumeration for possible task types
 */
export enum ContentStatusType {
  all = 'all',
  active = 'active',
  downloading = 'downloading',
  paused = 'paused',
  finished = 'finished',
  finishing = 'finishing',
  error = 'error',
}

export type ContentTypeId<T = Content['id']> = { source: ContentSource; key: Content['key']; id: T };
export type ContentStatusTypeId<T = ContentTypeId<Content['id']>> = Record<ContentStatusType, Set<T>>;

export interface ContentCount {
  badge: number;
  total: number;
  tabs: TabCount;
}
