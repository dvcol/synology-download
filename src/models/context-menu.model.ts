import { v4 as uuid } from 'uuid';

/**
 * Options for saved context Menus
 */
export interface ContextMenuOption {
  id: string;
  title: string;
  contexts: string[];
}

export const defaultMenu: ContextMenuOption = {
  id: uuid(),
  title: 'Download with Synology Diskstation',
  contexts: ['link', 'audio', 'video', 'image', 'selection'],
};
