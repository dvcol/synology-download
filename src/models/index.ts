export * from './context-menu.model';
export * from './http.model';
export * from './material-ui.model';
export * from './message.model';
export * from './navbar.model';
export * from './settings.model';
export * from './synology.model';
export * from './tab.model';
export * from './task.model';
export * from './tasks.model';
