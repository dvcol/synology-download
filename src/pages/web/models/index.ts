export * from './components.model';
export * from './fetch.model';
export * from './patch-options.model';
export * from './locales.model';
export * from './mock.model';
