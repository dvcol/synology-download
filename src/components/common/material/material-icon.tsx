import { jsx } from '@emotion/react';

import AddIcon from '@mui/icons-material/Add';
import DownloadIcon from '@mui/icons-material/Download';
import FileCopyIcon from '@mui/icons-material/FileCopy';
import FolderIcon from '@mui/icons-material/Folder';
import InsertPhotoIcon from '@mui/icons-material/InsertPhoto';
import LibraryMusicIcon from '@mui/icons-material/LibraryMusic';
import MusicNoteIcon from '@mui/icons-material/MusicNote';
import PhotoLibraryIcon from '@mui/icons-material/PhotoLibrary';
import VideogameAssetIcon from '@mui/icons-material/PhotoLibrary';
import SmartDisplayIcon from '@mui/icons-material/SmartDisplay';
import StorageIcon from '@mui/icons-material/Storage';
import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
import VpnKeyIcon from '@mui/icons-material/VpnKey';
import { SvgIconProps } from '@mui/material';

import React from 'react';

import { MaterialIcon } from '@src/models';

import JSX = jsx.JSX;

export const MuiIcon = ({ icon, props }: { icon: MaterialIcon; props?: SvgIconProps }): JSX.Element => {
  switch (icon) {
    case MaterialIcon.add:
      return <AddIcon />;
    case MaterialIcon.music:
      return <MusicNoteIcon />;
    case MaterialIcon.musicLibrary:
      return <LibraryMusicIcon />;
    case MaterialIcon.video:
      return <SmartDisplayIcon />;
    case MaterialIcon.videoLibrary:
      return <LibraryMusicIcon />;
    case MaterialIcon.photo:
      return <InsertPhotoIcon />;
    case MaterialIcon.photoLibrary:
      return <PhotoLibraryIcon />;
    case MaterialIcon.document:
      return <FileCopyIcon />;
    case MaterialIcon.game:
      return <VideogameAssetIcon />;
    case MaterialIcon.secure:
      return <VpnKeyIcon />;
    case MaterialIcon.private:
      return <VisibilityOffIcon />;
    case MaterialIcon.server:
      return <StorageIcon />;
    case MaterialIcon.folder:
      return <FolderIcon />;
    case MaterialIcon.download:
      return <DownloadIcon {...props} />;
    default:
      return <></>;
  }
};
