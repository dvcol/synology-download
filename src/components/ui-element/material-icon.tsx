import React from 'react';
import { MaterialIcon } from '../../models';
import { SvgIconProps } from '@mui/material';
import { jsx } from '@emotion/react';

import DownloadIcon from '@mui/icons-material/Download';
import AddIcon from '@mui/icons-material/Add';
import FolderIcon from '@mui/icons-material/Folder';
import MusicNoteIcon from '@mui/icons-material/MusicNote';
import LibraryMusicIcon from '@mui/icons-material/LibraryMusic';
import SmartDisplayIcon from '@mui/icons-material/SmartDisplay';
import InsertPhotoIcon from '@mui/icons-material/InsertPhoto';
import PhotoLibraryIcon from '@mui/icons-material/PhotoLibrary';
import VideogameAssetIcon from '@mui/icons-material/PhotoLibrary';
import FileCopyIcon from '@mui/icons-material/FileCopy';
import VpnKeyIcon from '@mui/icons-material/VpnKey';
import VisibilityOffIcon from '@mui/icons-material/VisibilityOff';
import StorageIcon from '@mui/icons-material/Storage';
import JSX = jsx.JSX;

export const MuiIcon = ({ icon, props }: { icon: MaterialIcon; props?: SvgIconProps }): JSX.Element => {
  switch (icon) {
    case MaterialIcon.add:
      return <AddIcon />;
    case MaterialIcon.music:
      return <MusicNoteIcon />;
    case MaterialIcon.musicLibrary:
      return <LibraryMusicIcon />;
    case MaterialIcon.video:
      return <SmartDisplayIcon />;
    case MaterialIcon.videoLibrary:
      return <LibraryMusicIcon />;
    case MaterialIcon.photo:
      return <InsertPhotoIcon />;
    case MaterialIcon.photoLibrary:
      return <PhotoLibraryIcon />;
    case MaterialIcon.document:
      return <FileCopyIcon />;
    case MaterialIcon.game:
      return <VideogameAssetIcon />;
    case MaterialIcon.secure:
      return <VpnKeyIcon />;
    case MaterialIcon.private:
      return <VisibilityOffIcon />;
    case MaterialIcon.server:
      return <StorageIcon />;
    case MaterialIcon.folder:
      return <FolderIcon />;
    case MaterialIcon.download:
      return <DownloadIcon {...props} />;
    default:
      return <></>;
  }
};
