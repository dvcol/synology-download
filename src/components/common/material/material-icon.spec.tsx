import React from 'react';

import { render, RenderResult } from '@testing-library/react';
import { MaterialIcon } from '@src/models';
import { MuiIcon } from './material-icon';

describe('material-icon.tsx component', () => {
  Object.values(MaterialIcon).forEach((icon) => {
    it(`Renders ${icon} icon`, () => {
      const app: RenderResult = render(<MuiIcon icon={icon} />);
      expect(app.asFragment()).toMatchSnapshot(icon);
    });
  });
});
