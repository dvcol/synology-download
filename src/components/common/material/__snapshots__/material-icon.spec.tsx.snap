// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`material-icon.tsx component Renders add icon: add 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="AddIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders document icon: document 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="FileCopyIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M16 1H4c-1.1 0-2 .9-2 2v14h2V3h12V1zm-1 4 6 6v10c0 1.1-.9 2-2 2H7.99C6.89 23 6 22.1 6 21l.01-14c0-1.1.89-2 1.99-2h7zm-1 7h5.5L14 6.5V12z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders download icon: download 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="DownloadIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M5 20h14v-2H5v2zM19 9h-4V3H9v6H5l7 7 7-7z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders folder icon: folder 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="FolderIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M10 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V8c0-1.1-.9-2-2-2h-8l-2-2z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders game icon: game 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="PhotoLibraryIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4 2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders music icon: music 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="MusicNoteIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 3v10.55c-.59-.34-1.27-.55-2-.55-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4V7h4V3h-6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders musicLibrary icon: musicLibrary 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="LibraryMusicIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 5h-3v5.5c0 1.38-1.12 2.5-2.5 2.5S10 13.88 10 12.5s1.12-2.5 2.5-2.5c.57 0 1.08.19 1.5.51V5h4v2zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders photo icon: photo 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="InsertPhotoIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M21 19V5c0-1.1-.9-2-2-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2zM8.5 13.5l2.5 3.01L14.5 12l4.5 6H5l3.5-4.5z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders photoLibrary icon: photoLibrary 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="PhotoLibraryIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M22 16V4c0-1.1-.9-2-2-2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2zm-11-4 2.03 2.71L16 11l4 5H8l3-4zM2 6v14c0 1.1.9 2 2 2h14v-2H4V6H2z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders private icon: private 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="VisibilityOffIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12 7c2.76 0 5 2.24 5 5 0 .65-.13 1.26-.36 1.83l2.92 2.92c1.51-1.26 2.7-2.89 3.43-4.75-1.73-4.39-6-7.5-11-7.5-1.4 0-2.74.25-3.98.7l2.16 2.16C10.74 7.13 11.35 7 12 7zM2 4.27l2.28 2.28.46.46C3.08 8.3 1.78 10.02 1 12c1.73 4.39 6 7.5 11 7.5 1.55 0 3.03-.3 4.38-.84l.42.42L19.73 22 21 20.73 3.27 3 2 4.27zM7.53 9.8l1.55 1.55c-.05.21-.08.43-.08.65 0 1.66 1.34 3 3 3 .22 0 .44-.03.65-.08l1.55 1.55c-.67.33-1.41.53-2.2.53-2.76 0-5-2.24-5-5 0-.79.2-1.53.53-2.2zm4.31-.78 3.15 3.15.02-.16c0-1.66-1.34-3-3-3l-.17.01z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders secure icon: secure 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="VpnKeyIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M12.65 10C11.83 7.67 9.61 6 7 6c-3.31 0-6 2.69-6 6s2.69 6 6 6c2.61 0 4.83-1.67 5.65-4H17v4h4v-4h2v-4H12.65zM7 14c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders server icon: server 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="StorageIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M2 20h20v-4H2v4zm2-3h2v2H4v-2zM2 4v4h20V4H2zm4 3H4V5h2v2zm-4 7h20v-4H2v4zm2-3h2v2H4v-2z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders video icon: video 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="SmartDisplayIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 4H4c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zM9.5 16.5v-9l7 4.5-7 4.5z"
    />
  </svg>
</DocumentFragment>
`;

exports[`material-icon.tsx component Renders videoLibrary icon: videoLibrary 1`] = `
<DocumentFragment>
  <svg
    aria-hidden="true"
    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
    data-testid="LibraryMusicIcon"
    focusable="false"
    viewBox="0 0 24 24"
  >
    <path
      d="M20 2H8c-1.1 0-2 .9-2 2v12c0 1.1.9 2 2 2h12c1.1 0 2-.9 2-2V4c0-1.1-.9-2-2-2zm-2 5h-3v5.5c0 1.38-1.12 2.5-2.5 2.5S10 13.88 10 12.5s1.12-2.5 2.5-2.5c.57 0 1.08.19 1.5.51V5h4v2zM4 6H2v14c0 1.1.9 2 2 2h14v-2H4V6z"
    />
  </svg>
</DocumentFragment>
`;
